ledger/participant-integration-api/src/test/lib/scala/platform/store/dao/JdbcLedgerDaoContractsSpec.scala

```diff
@@ -9,25 +9,88 @@ import java.util.UUID
 import com.daml.lf.transaction.GlobalKey
 import com.daml.lf.transaction.Node.KeyWithMaintainers
 import com.daml.lf.value.Value.{ContractId, ContractInst, ValueText}
-import org.scalatest.{Inside, LoneElement, OptionValues}
+import com.daml.platform.store.dao.events.contracts.LedgerDaoContractsReader
+import com.daml.platform.store.dao.events.contracts.LedgerDaoContractsReader.KeyUnassigned
 import org.scalatest.flatspec.AsyncFlatSpec
 import org.scalatest.matchers.should.Matchers
+import org.scalatest.{Inside, LoneElement, OptionValues}
 
 private[dao] trait JdbcLedgerDaoContractsSpec extends LoneElement with Inside with OptionValues {
   this: AsyncFlatSpec with Matchers with JdbcLedgerDaoSuite =>
 
+  private def contractsReader = ledgerDao.contractsReader
+
   behavior of "JdbcLedgerDao (contracts)"
 
   it should "be able to persist and load contracts" in {
     for {
       (_, tx) <- store(singleCreate)
-      result <- ledgerDao.lookupActiveOrDivulgedContract(nonTransient(tx).loneElement, Set(alice))
+      result <- contractsReader.lookupActiveContractAndLoadArgument(
+        nonTransient(tx).loneElement,
+        Set(alice),
+      )
     } yield {
       // The agreement text is always empty when retrieved from the contract store
       result shouldEqual Some(someVersionedContractInstance.copy(agreementText = ""))
     }
   }
 
+  it should "present the contract state at a specific event sequential id" in {
+    for {
+      (_, tx) <- store(singleCreate(create(_, signatories = Set(alice))))
+      contractId = nonTransient(tx).loneElement
+      _ <- store(singleNonConsumingExercise(contractId))
+      eventSeqIdAtCreate <- ledgerDao.lookupLedgerEndSequentialId()
+      _ <- store(txArchiveContract(alice, (contractId, None)))
+      eventSeqIdAfterArchive <- ledgerDao.lookupLedgerEndSequentialId()
+      queryAfterCreate <- contractsReader.lookupContractState(contractId, eventSeqIdAtCreate)
+      queryAfterArchive <- contractsReader.lookupContractState(contractId, eventSeqIdAfterArchive)
+    } yield {
+      queryAfterCreate.value match {
+        case LedgerDaoContractsReader.ActiveContract(contract, stakeholders, _) =>
+          contract shouldBe someVersionedContractInstance.copy(agreementText = "")
+          stakeholders should contain theSameElementsAs Set(alice)
+        case LedgerDaoContractsReader.ArchivedContract(_) =>
```

Comment:
Minor: why not use just `case _ =>`? Or fail with information about the matched contract type - `ArchivedContract` in this case.